### PR TITLE
VIT Decoder updates

### PR DESCRIPTION
**Pull request by benjijamorris**
## What does this PR do?
- add CrossMAE
- Fix supperres typo
- add multi-channel projection convolutional decoder option

## Before submitting

- [x] Did you make sure **title is self-explanatory** and **the description concisely explains the PR**?
- [] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **test your PR locally** with `pytest` command?
- [x] Did you **run pre-commit hooks** with `pre-commit run -a` command?

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: cyto_dl/nn/vits/blocks/cross_attention.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        num_heads=8,
+        qkv_bias=False,
+        qk_scale=None,
+        attn_drop=0.0,
+        proj_drop=0.0,
+    ):
+        super().__init__()
+        self.num_heads = num_heads
+        head_dim = decoder_dim // num_heads
+        # NOTE scale factor was wrong in my original version, can set manually to be compat with prev weights
```
Comment (ritvikvasan) [inline cyto_dl/nn/vits/blocks/cross_attention.py:44]: is this your note?

File: cyto_dl/nn/vits/seg.py [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+import numpy as np
 import torch
 from einops.layers.torch import Rearrange
 from monai.networks.blocks import UnetOutBlock, UnetResBlock, UpSample
 
 from cyto_dl.nn.vits.mae import MAE_Encoder
 
 
-class SupperresDecoder(torch.nn.Module):
+class SuperresDecoder(torch.nn.Module):
```
Comment (ritvikvasan) [inline cyto_dl/nn/vits/seg.py:11]: this is unrelated to crossMAE but just in this PR?

**Review by ritvikvasan — DISMISSED** [review]

File: cyto_dl/nn/vits/blocks/cross_attention.py (hunk 1 above)
Comment (benjijamorris) [reply cyto_dl/nn/vits/blocks/cross_attention.py:44]: no, I can remove that

File: cyto_dl/nn/vits/seg.py (hunk 2 above)
Comment (benjijamorris) [reply cyto_dl/nn/vits/seg.py:11]: oops, started as a general decoder update, I can rename the pr

**Pull request merged**